### PR TITLE
Add decision: IdP group membership mapping

### PR DESCRIPTION
**Pull request by ccremer**
## Checklist

- [x] Try to isolate changes into separate PRs (to build a better changelog).
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `change`, `decision`, `requirement/quality`, `requirement/functional`, `dependency`
      as they show up in the changelog
- [x] Link this PR to related issues if applicable.

### CODE REVIEW

**Comment by ccremer** [comment]
Please also check for terminology, I'm not really an OIDC expert.

File: docs/antora.yml [hunk 1]
```diff
@@ -8,3 +8,4 @@ asciidoc:
   attributes:
     product: APPUiO Cloud
     zone: APPUiO Zone
+    idp: APPUiO Keycloak
```
Comment (ccremer) [inline docs/antora.yml:11]: Any preferences here? Other alternative would be `APPUiO IDP` or `APPUiO ID`. Other proposals?

Comment (tobru) [reply docs/antora.yml:11]:
```suggestion
    idp: APPUiO IdP
```

Note: The `d` should be lowercase, see https://csrc.nist.gov/glossary/term/Identity_Provider.

File: docs/modules/ROOT/partials/nav-explanation.adoc [hunk 2]
```diff
@@ -5,6 +5,7 @@
 
 * Decisions
 ** xref:appuio-cloud:ROOT:explanation/decisions/keycloak.adoc[Keycloak as IDP]
+** xref:appuio-cloud:ROOT:explanation/decisions/idp-group-mapping.adoc[IDP group membership mapping]
```
Comment (tobru) [inline docs/modules/ROOT/partials/nav-explanation.adoc:8]:
```suggestion
** xref:appuio-cloud:ROOT:explanation/decisions/idp-group-mapping.adoc[IdP group membership mapping]
```

File: docs/modules/ROOT/pages/explanation/decisions/idp-group-mapping.adoc [hunk 3]
```diff
@@ -0,0 +1,52 @@
+= IDP Group membership Mapping
+
+== Problem
+
+Normal users can login via Keycloak to any {zone} and deploy workloads.
+However, certain groups are destined to be cluster administrators and require the permissions to perform privileged operations.
+
+With Keycloak being the only IDP frontend for any {zone}, group memberships need to be synchronized in a way that OpenShift is able to distinguish special organizations.
```
Comment (tobru) [inline docs/modules/ROOT/pages/explanation/decisions/idp-group-mapping.adoc:8]:
```suggestion
With Keycloak being the only IdP frontend for any {zone}, group memberships need to be synchronized in a way that OpenShift is able to distinguish special organizations.
```

File: docs/modules/ROOT/pages/explanation/decisions/idp-group-mapping.adoc [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+Especially with brokered IDPs where the actual source of truth regarding group memberships of users is outside of {idp} this is a technical challenge around authorization that impacts the architecture.
```
Comment (tobru) [inline docs/modules/ROOT/pages/explanation/decisions/idp-group-mapping.adoc:9]:
```suggestion
Especially with brokered IdPs where the actual source of truth regarding group memberships of users is outside of {idp} this is a technical challenge around authorization that impacts the architecture.
```

File: docs/modules/ROOT/pages/explanation/decisions/idp-group-mapping.adoc [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+The problem ultimately asks the questions:
+* How does {idp} know which organization is privileged if the organization is brokered through another IDP?
```
Comment (tobru) [inline docs/modules/ROOT/pages/explanation/decisions/idp-group-mapping.adoc:12]:
```suggestion
* How does {idp} know which organization is privileged if the organization is brokered through another IdP?
```

File: docs/modules/ROOT/pages/explanation/decisions/idp-group-mapping.adoc [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+* How does {idp} determine if a user, whose login is brokered though another IDP, belongs to such an organization?
```
Comment (tobru) [inline docs/modules/ROOT/pages/explanation/decisions/idp-group-mapping.adoc:13]:
```suggestion
* How does {idp} determine if a user, whose login is brokered though another IdP, belongs to such an organization?
```

File: docs/modules/ROOT/pages/explanation/decisions/idp-group-mapping.adoc [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+== Relevant requirements
+
+* xref:references/quality-requirements/security/impersonating-cluster-admin.adoc[Impersonating cluster-admin]
+
+== Relevant decisions
+
+* xref:explanation/decisions/keycloak.adoc[Keycloak as IDP]
```
Comment (tobru) [inline docs/modules/ROOT/pages/explanation/decisions/idp-group-mapping.adoc:21]:
```suggestion
* xref:explanation/decisions/keycloak.adoc[Keycloak as IdP]
```

File: docs/modules/ROOT/pages/explanation/decisions/idp-group-mapping.adoc [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+== Alternatives
+
+[qanda]
+External group syncing via API::
+An extra component connects to the brokered IDP via API, collects the groups and group memberships, adapts and then pushes to {idp} also via API.
+This method allows different implementations for various brokered IDPs, for example there could be an implementation for Keycloak-to-Keycloak, or GitHub-to-Keycloak.
```
Comment (tobru) [inline docs/modules/ROOT/pages/explanation/decisions/idp-group-mapping.adoc:28]:
```suggestion
This method allows different implementations for various brokered IdPs, for example there could be an implementation for Keycloak-to-Keycloak, or GitHub-to-Keycloak.
```

File: docs/modules/ROOT/pages/explanation/decisions/idp-group-mapping.adoc [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+There is a https://github.com/amomra/oidc-group-mapper[GitHub project] that implements this approach as a Keycloak extension written in Java, but it seems to be unmaintained.
+
+== Decision
+
+Group mapping via claims
+
+== Rationale
+
+A mapping via claims basically allows to delegate the desire and support of group memberships to the implementation of the brokered IDP.
+In other words, if the foreign IDP provides the group membership in the claims, then {idp} can sync the groups.
```
Comment (tobru) [inline docs/modules/ROOT/pages/explanation/decisions/idp-group-mapping.adoc:47]:
```suggestion
In other words, if the foreign IdP provides the group membership in the claims, then {idp} can sync the groups.
```

**Review by tobru — APPROVED** [review]
Besides the `IDP` vs `IdP` issue: All fine by me.

**Review by corvus-ch — APPROVED** [review]

**Review by glrf — APPROVED** [review]

**Pull request merged**